Assets/MRTK/SDK/Experimental/Features/UX/BoundsControl/BoundsControl.cs

```diff
@@ -982,12 +1025,23 @@ private void TransformTarget(HandleType transformType)
                         }
                     }
 
-                    Target.transform.localScale = clampedTransform.Scale;
-                    Target.transform.position = initialPositionOnGrabStart * scaleFactor + (1 - scaleFactor) * oppositeCorner;
+                    var newPosition = initialPositionOnGrabStart * scaleFactor + (1 - scaleFactor) * oppositeCorner;
+                    Target.transform.localScale = SmoothTo(Target.transform.localScale, clampedTransform.Scale, scaleLerpTime);
+                    Target.transform.position = SmoothTo(Target.transform.position, newPosition, scaleLerpTime);
                 }
             }
         }
 
+        private Vector3 SmoothTo(Vector3 source, Vector3 goal, float lerpTime)
+        {
+            return Vector3.Lerp(source, goal, (!smoothingActive || lerpTime == 0f) ? 1f : 1f - Mathf.Pow(lerpTime, Time.deltaTime));
```

Comment:
We should probably break the SmoothTo function to some other internal class that can be used between BoundsControl and ObjectManipulator (i.e. so not something that is accessible to external because it's somewhat of an implementation detail to these other classes and we don't necessarily want to maintain this specific surface of the API going forward).

In general we should copy paste and put things in to Utils or something else where code reuse makes sense.

Reply:
I agree.

I was contemplating this in relation to the elastic system. The elastic system is arguably another kind of "smoothing" function, in that it takes a source, goal, and delta time. The only difference is that one is stateful, and one is not. (And that if you'd like the object to be springy even after you let go, you would need to add some code to the Update() function.)

Ideally we could have a generic way of addressing both. Although I figured that was sort of out of scope for this PR.

I'm intending on bringing the elastic/magnetic functionality to BoundsControl (and hopefully ObjectManipulator so that will be a good time to merge this functionality, I think.)